agent/pkg/singularity/singularity.go

```diff
@@ -288,15 +289,9 @@ func (s *SingularityClient) RunContainer(
 		}
 	}
 
-	// TODO(DET-9075): Un-dockerize the RunContainer API so we can know to pass `--rocm` without
-	// regexing on devices.
 	// TODO(DET-9080): Test this on ROCM devices.
-	rocmDevice := regexp.MustCompile("/dev/dri/by-path/pci-.*-card")
-	for _, d := range req.HostConfig.Devices {
-		if rocmDevice.MatchString(d.PathOnHost) {
-			args = append(args, "--rocm")
-			break
-		}
+	if s.slotType == "rocm" {
+		args = append(args, "--rocm")
```

Comment:
slot type is configured by the job/resource pool not the agent, i think this would even force aux jobs to run with ROCM on agents configured this way